api/proto/teleport/legacy/types/types.proto

```diff
@@ -3064,6 +3064,17 @@ message AccessRequestSpecV3 {
 
   // DryRunEnrichment contains the extra info added in response to a dry run request.
   AccessRequestDryRunEnrichment DryRunEnrichment = 23 [(gogoproto.jsontag) = "dry_run_enrichment,omitempty"];
+
+  // RequestKind indicates the kind (short/long-term) of request.
+  AccessRequestKind RequestKind = 24 [(gogoproto.jsontag) = "request_kind,omitempty"];
+}
+
+enum AccessRequestKind {
+  // SHORT_TERM is the default value, and represents a short-term request,
+  // either role-based or resource-based.
+  SHORT_TERM = 0;
```

Comment:
Should zero-value maybe indicate the "undefined" request type for backwards compatibility or do you not foresee this being a problem? How would the reviewer UI handle access requests created prior to this change otherwise?

Reply:
That's actually a good question, what should happen to "legacy" access requests that don't have this flag set?

I'm thinking that for a more graceful migration, we should keep the existing behavior/UI for access requests created before this change, and only show the new long-term or short-term specific UI for new access requests, and then phase out the old UI in the next major version. So in v18 if there's an access request that was created before upgrade to this but hasn't expired yet, it will use the current UI, new access requests will use the new short/long-term specific ones. In v19 we can remove the old reviewer view.

For this, you do need the 0 value to be "undefined".

WDYT?

Reply:
Hm, could do, to support that. That makes sense to me (although, I'm not sure there are any differences in functionality between the new short-term UI and existing UI? both still allow promoting to an access list iirc)